### PR TITLE
use ujson/cStringIO/python-lzf if they're available

### PR DESCRIPTION
**Pull request by joshowen**

### CODE REVIEW

**Comment by joshowen** [comment]
@sripathikrishnan thoughts?

**Comment by joshowen** [comment]
@sripathikrishnan  is this maintained anymore?

**Comment by MOON-CLJ** [comment]
vote for this pr

File: rdbtools/parser.py [hunk 1]
```diff
@@ -653,38 +662,41 @@ def get_logical_type(self, data_type):
...
-            else :
-                length = ctrl >> 5
-                if length == 7 :
-                    length = length + in_stream[in_index]
-                    in_index = in_index + 1
-                
-                ref = out_index - ((ctrl & 0x1f) << 8) - in_stream[in_index] - 1
+        if HAS_PYTHON_LZF:
+            return lzf.decompress(compressed, expected_length)
+        else:
```
Comment (MOON-CLJ) [inline rdbtools/parser.py]: no need for this "else:", this code below will not running because of return。

**Comment by MOON-CLJ** [comment]
use python-lzf intead, can also gain big benefits on performance。as shown below,

```
real    30m51.416s
user    30m44.560s
sys     0m6.068s
```
vs

``` 
real    7m34.941s
user    7m25.084s
sys     0m5.480s
```

cons:
if  python-lzf upgrade to new verison lzf but redis dont, it may or may not have some problem。

**Comment by joshowen** [comment]
@oranagra any interest in merging this PR?  I can update rebase it to master if you want.

**Comment by oranagra** [comment]
@joshowen yeah I don't see any reason not to merge it, but i see that @mihu recently merged many things to his fork, maybe we can utilize the merging work he already did.

**Comment by joshowen** [comment]
Sounds good.  Ping me if you want me to rebase.

**Pull request merged**